src/data/actions/courseInfo.js

```diff
@@ -50,3 +67,18 @@ export function fetchCourseInfo(id) {
       });
   };
 }
+
+export function createCourse(courseData) {
+  return (dispatch) => {
+    dispatch(createNewCourse(courseData));
+    // Send create course POST
+    return DiscoveryDataApiService.createCourse(courseData)
+      .then((response) => {
+        const course = response.data;
+        dispatch(courseCreateSuccess(course));
+      })
+      .catch((error) => {
+        dispatch(courseCreateFail(`Course create failed, please try again or contact support. Error( ${error.response.data} )`));
+      });
+  };
+}
```

Comment:
Guh! All this boilerplate for the same idea. "Do a request, handle request success/fail" - the fedx recommendation was just writing a factory class for these. We should look at that at some point. For now, it's just easier to copy and paste. But really.